pages/price-feeds/use-real-time-data/solana.mdx

```diff
@@ -1,23 +1,149 @@
-# Pyth on Solana
+---
+description: Consume Pyth Network prices in Solana applications
+---
 
-We provide one SDK for Solana on-chain programs to use Pyth prices:
+# How to Use Real-Time Data in Solana Programs
 
-The [pyth-sdk-solana crate](https://github.com/pyth-network/pyth-sdk-rs/tree/main/pyth-sdk-solana) can be used to consume Pyth prices inside Solana programs written in Rust.
+This guide explains how to use real-time Pyth data in Solana applications.
 
-## Example Program
+## Install Pyth SDKs
 
-The Pyth Solana SDK includes two example programs that demonstrate how to use Pyth price feeds in Solana programs:
-a [program using only the basic Solana SDK](https://github.com/pyth-network/pyth-sdk-rs/tree/main/examples/sol-contract) and an [Anchor program](https://github.com/pyth-network/pyth-sdk-rs/tree/main/examples/sol-anchor-contract).
+We provide two SDKs for Solana applications to cover the on- and off-chain portions of the integration:
 
-## Price Feed IDs
+### Rust SDK
 
-The price feeds available on each Solana network are listed on the following pages:
+The [pyth-solana-receiver-sdk crate](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/solana/pyth_solana_receiver_sdk) can be used to consume Pyth prices inside Solana programs written in Rust.
+Add this crate to the dependencies section of your `Cargo.toml` file:
 
-- [Devnet](https://pyth.network/developers/price-feed-ids#solana-devnet)
-- [Testnet](https://pyth.network/developers/price-feed-ids#solana-testnet)
-- [Mainnet-Beta](https://pyth.network/developers/price-feed-ids#solana-mainnet-beta)
+```toml copy
+[dependencies]
+pyth-solana-receiver-sdk ="0.1.0"
+```
 
-## Program Addresses
+### Typescript SDK
 
-Developers will need the address of the Pyth oracle πrogram on their blockchain in order to use Pyth.
-Please consult [Solana Contract Addresses](../contract-addresses/solana) to find the address for your blockchain.
+Pyth provides two Typescript packages, [@pythnetwork/price-service-client](https://github.com/pyth-network/pyth-crosschain/tree/main/price_service/client/js) and [@pythnetwork/pyth-solana-receiver](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/solana/sdk/js/pyth_solana_receiver), for fetching Pyth prices and submitting them to the blockchain respectively.
+Add these packages to your off-chain dependencies:
+
+```bash copy
+npm install --save @pythnetwork/price-service-client @pythnetwork/pyth-solana-receiver
+```
+
+## Write Contract Code
+
+Add code to your Solana program to read the Pyth price.
+Pyth prices are posted to price update accounts that can be passed to any instruction that needs price data.
+If you are using Anchor, you can simply add an `Account<'info, PriceUpdateV2>` field to your `Context` struct:
+
+TODO: does this Account thing perform the necessary account ownership checks?
```

Comment:
@guibescos there's some program ownership check that needs to be added for these accounts right?

Reply:
The `Account` struct does the ownership check automatically

Reply:
We'll probably a non-anchor way to do this